doc/develop/api/api_conventions.rst

```diff
@@ -0,0 +1,49 @@
+.. _api_conventions.rst:
+
+API Conventions
+###############
+
+Zephyr provides a few programming interfaces that users can use for creating their applications.
+Zephyr's API space includes the following areas:
+- C programming interface (function prototypes, structures, and macros), provided in C header files
+- Configuration system (Kconfig)
+- Device driver model (Devicetree)
+
+Other areas, such as the build system (CMake), may also be seen as Zephyr's API. However, any area
+that is not explicitly listed herein is not considered Zephyr's API as of today.
+
+API classification
+******************
+The files that define the Zephyr APIs may contain symbols intended for different usage. The intended
+API usage is designated by the API class. Zephyr APIs are classified as:
+
+- private - these APIs are intended for use in-tree within boundary of a software component such as
```

Comment:
API usage and API class is not limited to just within the Zephyr repository.  Please:

- Remove "in-tree" from the definition so that it is a broadly applicable definition;
- Qualify the lifecycle statement so that it applies to only those APIs defined in the Zephyr repository.

Reply:
Done.